### PR TITLE
XML read/write

### PR DESCRIPTION
**Pull request by danmoseley**
Remove expensive file existence checks - it will throw if they don't exist.
Read docs xml files in the correct codepage (it chokes eg on Char.xml otherwise). We already save in 1252.
Include file path when failing to read/write the XML.

### CODE REVIEW

File: Libraries/Docs/DocsCommentsContainer.cs [hunk 1]
```diff
@@ -174,12 +174,18 @@ private IEnumerable<FileInfo> EnumerateFiles()
...
         private void LoadFile(FileInfo fileInfo)
         {
-            if (!fileInfo.Exists)
+            try
             {
-                throw new Exception($"Docs xml file does not exist: {fileInfo.FullName}");
+                // docs repo uses code page 1252
+                Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+                StreamReader sw = new(fileInfo.FullName, Encoding.GetEncoding(1252));
+                xDoc = XDocument.Load(sw);
```
Comment (jeffhandley) [inline Libraries/Docs/DocsCommentsContainer.cs:182]: Nit:

```suggestion
                StreamReader sr = new(fileInfo.FullName, Encoding.GetEncoding(1252));
                xDoc = XDocument.Load(sr);
```

File: Libraries/IntelliSenseXml/IntelliSenseXmlCommentsContainer.cs [hunk 2]
```diff
@@ -85,12 +86,15 @@ private IEnumerable<FileInfo> EnumerateFiles()
...
         {
-            if (!fileInfo.Exists)
+            try
             {
-                throw new Exception($"The IntelliSense xml file does not exist: {fileInfo.FullName}");
+                xDoc = XDocument.Load(fileInfo.FullName);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(string.Format("Failed to load '{0}'. {1}", fileInfo.FullName, ex.ToString()));
```
Comment (jeffhandley) [inline Libraries/IntelliSenseXml/IntelliSenseXmlCommentsContainer.cs:95]: Suggestion:

```suggestion
                Log.Error($"Failed to load '{fileInfo.FullName}'. {ex}");
```

**Review by jeffhandley — APPROVED** [review]
Just a couple of nit-picky suggestions.

File: Libraries/IntelliSenseXml/IntelliSenseXmlCommentsContainer.cs (hunk 2 above)
Comment (danmoseley) [reply Libraries/IntelliSenseXml/IntelliSenseXmlCommentsContainer.cs]: Ha, sometimes my C# regresses to circa 2005.

**Review by jeffhandley — APPROVED** [review]

**Pull request merged**